instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/ServiceLoaderUtil.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.function.Function;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class ServiceLoaderUtil {
+
+  private static Function<Class<?>, List<?>> loaderFunction =
```

Comment:
I'd go with `Function<Class<?>, Iterable<?>>` so it could be initialized with just `ServiceLoader::load;`. The field could also be `volatile`